pennylane/devices/default_qubit.py

```diff
@@ -484,6 +498,11 @@ def expval(self, observable, shot_range=None, bin_size=None):
 
         Returns:
             float: returns the expectation value of the observable
+
+        .. warning::
+
+            This function only supports broadcasting if no differentiation with
+            respect to the Hamiltonian or the quantum state directly is performed.
```

Comment:
I could be wrong, but my reading of `backprop_mode` logic below is:

- There is a Hamiltonian measured
- The user has selected `default.qubit`
- The interface is _not_ `None`
- `shots=None`.

That is, the `backprop_mode` branch can be selected during a _forward_ pass even if the Hamiltonian/state is not being directly differentiated?

Reply:
In which case, this message should be updated to say that 'broadcasting is not supported when `shots=None` and Hamiltonians are being measured'?

Reply:
Thanks @dwierichs! Yeah it took me a moment to work out the logic as well, but the `not isinstance(self.state, np.ndarray)` check will catch forward passes of JAX/Torch/TF